tests/core/test_typedefs.py

```diff
@@ -38,3 +51,27 @@ def test_output_has_changes():
 
 def test_output_has_call_in():
     assert Optional[SimTime] == Output.__annotations__["call_at"]
+
+
+def test_Interrupt_is_dataclass():
+    assert is_dataclass(Interrupt)
+
+
+def test_Interrupt_has_source():
+    assert ComponentID == Interrupt.__annotations__["source"]
+
+
+def test_ComponentException_is_dataclass():
+    assert is_dataclass(ComponentException)
+
+
+def test_ComponentException_has_source():
+    assert ComponentID == ComponentException.__annotations__["source"]
+
+
+def test_ComponentException_has_error():
+    assert Exception == ComponentException.__annotations__["error"]
+
+
+def test_ComponentException_has_traceback():
```

Comment:
Nitpick: this should still be called `test_component_exception_has_traceback`. Function naming convention wins in this case.

Reply:
Yep, my bad